lib/irb/init.rb

```diff
@@ -309,16 +313,22 @@ def IRB.parse_opts(argv: ::ARGV)
         IRB.print_usage
         exit 0
       when "--"
-        if opt = argv.shift
+        if !noscript && (opt = argv.shift)
           @CONF[:SCRIPT] = opt
           $0 = opt
         end
         break
+      when '-'
+        # nothing
       when /^-/
```

Comment:
Is this still needed with the above case?

Reply:
Also, maybe it's worth to cover `-` in tests too?

Reply:
I did some more testing, and `-` wasn't being handled as @nobu indicated it should be (see https://github.com/ruby/irb/pull/22#pullrequestreview-617190781).  I've pushed a commit to address that, so that `-` is treated as the standard input.